doc/source/debug.rst

```diff
@@ -0,0 +1,188 @@
+      tag:
+          ae885c0a6226
+
+and run a helm upgrade::
+
+    helm upgrade jhub jupyterhub/jupyterhub --version=v0.4 -f config.yaml
+
+.. note::
+
+   In this case our Kubernetes namespace is called ``jhub``. Yours may be
+   different depending on what you chose during setup.
+
+.. note::
+
+   Depending on the size of the Docker image, this may take a while to complete.
+
+Right after you run this command, let's once again list the pods in our
+deployment::
+
+  $ kubectl get pod --namespace=jhub
+  NAME                                READY     STATUS              RESTARTS   AGE
+  hub-deployment-2653507799-r7wf8     0/1       ContainerCreating   0          31s
+  hub-deployment-3311438805-xnfvp     1/1       Terminating         0          14m
+  jupyter-choldgraf                   0/1       ImagePullBackOff    0          12m
+  proxy-deployment-1227971824-mn2wd   1/1       Running             0          5h
+
+Here we can see one ``hub-deployment`` pod being destroyed, and another (based
+on the upgraded helm chart) being created. We also see our broken user pod,
+which will only be upgraded if it is restarted. To do this, let's delete the
+user's pod::
+
+    $ kubectl delete pod jupyter-choldgraf --namespace=jhub
+
+Finally, we'll tell our user to log back in to the JupyterHub. Then let's
+list our running pods once again::
+
+  $ kubectl get pod --namespace=jhub
+  NAME                                READY     STATUS    RESTARTS   AGE
+  hub-deployment-2653507799-r7wf8     1/1       Running   0          3m
+  jupyter-choldgraf                   1/1       Running   0          18s
+  proxy-deployment-1227971824-mn2wd   1/1       Running   0          5h
+
+And now we see that we have a running user pod!
+
+Note that many debugging situations are not as straightforward as this one.
+It will take some time before you get a feel for the errors that Kubernetes
+may throw at you, and how these are tied to your configuration files.
+
+Case studies in debugging Kubernetes
+------------------------------------
+The following sections contain some case studies that illustrate some of the
+more common bugs / gotchas that you may experience using JupyterHub with
+Kubernetes.
+
+Hub fails to start
+~~~~~~~~~~~~~~~~~~
+
+**Symptom:** following ``kubectl get pod``, the ``hub-deployment`` pod is in
+``Error`` or ``CrashLoopBackoff`` state, or appears to be running but is
+unreachable via the web.
+
+**Investigating:** the output of ``kubectl logs hub-deployment...`` shows something
+like::
+    File "/usr/local/lib/python3.5/dist-packages/jupyterhub/proxy.py", line 589, in get_all_routes
+      resp = yield self.api_request('', client=client)
+    tornado.httpclient.HTTPError: HTTP 403: Forbidden
+
+**Diagnosis:** This is likely because the ``hub-deployment`` pod cannot
+communicate with the proxy pod API, likely because of a problem in the
+``secretToken`` that was put in ``config.yaml``.
+
+**Fix:** Add to ``config.yaml``::

```

Comment:
May be better to have 3 steps in fix.

1. create a secret token
2. Edit config.yaml
3. Redeploy